Packs/AWS-CloudTrail/Integrations/AWS-CloudTrail/AWS-CloudTrail_test.py

```diff
@@ -0,0 +1,299 @@
+    "isMultiRegionTrail": "True",
+    "enableLogFileValidation": "True",
+    "cloudWatchLogsLogGroupArn": "",
+    "cloudWatchLogsRoleArn": "",
+    "kmsKeyId": "",
+}
+
+VALID_RESPONSE_METADATA = {"ResponseMetadata": {"HTTPStatusCode": 200}}
+
+
+@pytest.fixture
+def aws_cloudtrail():
+    return importlib.import_module("AWS-CloudTrail")
+
+
+@pytest.fixture(autouse=True)
+def setup(mocker):
+    mocker.patch.object(demisto, "params", return_value=PARAMS)
+
+
+@pytest.fixture
+def return_results_func(mocker, aws_cloudtrail):
+    return mocker.patch.object(aws_cloudtrail, "return_results")
+
+
+@pytest.fixture
+def return_error_func(mocker, aws_cloudtrail):
+    return mocker.patch.object(aws_cloudtrail, "return_error")
+
+
+class Config:
+    def __init__(self):
+        self._user_provided_options = {"region_name": PARAMS["defaultRegion"]}
+
+
+class CloudTrailClient:
+    def __init__(self) -> None:
+        self._client_config = Config()
+        self._trail = {
+            "Name": "name",
+            "S3BucketName": "name",
+            "IncludeGlobalServiceEvents": "true",
+            "IsMultiRegionTrail": "true",
+            "TrailARN": "trailARN",
+            "LogFileValidationEnabled": "true",
+            "SnsTopicName": "name",
+            "S3KeyPrefix": "prefix",
+            "SnsTopicARN": "arn",
+            "CloudWatchLogsLogGroupArn": "arn",
+            "CloudWatchLogsRoleArn": "arn",
+            "KmsKeyId": "id",
+            "HomeRegion": "region",
+        }
+
+    def create_trail(self, **kwargs):
+        return self._trail
+
+    def delete_trail(self, **kwargs):
+        return VALID_RESPONSE_METADATA
+
+    def describe_trails(self, **kwargs):
+        return {"trailList": [self._trail]} | VALID_RESPONSE_METADATA
+
+    def update_trail(self, **kwargs):
+        return self._trail
+
+    def start_logging(self, **kwargs):
+        return VALID_RESPONSE_METADATA
+
+    def stop_logging(self, **kwargs):
+        return VALID_RESPONSE_METADATA
+
+    def lookup_events(self, **kwargs):
+        return None
+
+    def get_paginator(self, _):
+        class Paginator:
+            def paginate(self, **kwargs):
+                return [{"Events": [{"Username": "user"}]}]
+        return Paginator()
+
+
+class STSClient:
+    def assume_role(self, **kwargs):
+        return {
+            "Credentials": {
+                "AccessKeyId": "1",
+                "SecretAccessKey": "2",
+                "SessionToken": "3",
+            },
+        }
+
+
+def mock_boto3_client(service_name, *args, **kwargs):
+    if service_name == 'sts':
+        return STSClient()
+    return CloudTrailClient()
+
+
+def test_aws_session_params(mocker, aws_cloudtrail):
+    """
+    Given
+    - demisto parameters
+    When
+    - Calling aws_session()
+    Then
+    - Ensure a cloudtrail client is returned
+    """
+    mocker.patch.object(aws_cloudtrail.boto3, "client", side_effect=mock_boto3_client)
+    aws_cloudtrail.aws_session()
```

Comment:
Are we asserting that no exception has been raised and the process was completed?

Reply:
added `assert` to ensure a non empty value is returned. the test will fail if an exception is raised